model/api_endpoint_lb.go

```diff
@@ -84,6 +102,10 @@ func (e APIEndpointLB) Validate() error {
 			return errors.New("hostedZone.id should not be specified when an API endpoint LB is not managed by kube-aws")
 		}
 
+		if e.Type != nil && len(*e.Type) > 0 {
```

Comment:
maybe also validate possible values here? accept `classic` or `network`, but nothing else

Reply:
If `id` is specified, we only validate that no `type` attribute is specified _at all_, which is the correct thing to do in case the load balancer `id` is specified (since we are reusing a ELB, there's no need to specify a type for it).

The check to verify the validity of `type` itself is performed down this same function:

https://github.com/kubernetes-incubator/kube-aws/pull/945/files/6af87179b66f5fd5da35cfabbd2fa7c9cb2db7d6#diff-971c12e3b00cffb1d084e6e1f5942103R136